Userland/Libraries/LibJS/Runtime/StringObject.cpp

```diff
@@ -61,11 +59,7 @@ static Optional<PropertyDescriptor> string_get_own_property(GlobalObject& global
     // NOTE: If the property name is a number type (An implementation-defined optimized
     // property key type), it can be treated as a string property that has already been
     // converted successfully into a canonical numeric index.
-    Value index;
-    if (property_name.is_string())
-        index = canonical_numeric_index_string(global_object, property_name.to_value(vm));
-    else
-        index = Value(property_name.as_number());
+    Value index = property_name.to_canonical_numeric_index_string(global_object);
```

Comment:
Please use `auto` where possible

Reply:
Used auto and removed comment about no longer visible optimization